projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DataJobsDeploymentController.java

```diff
@@ -51,14 +53,27 @@ public class DataJobsDeploymentController implements DataJobsDeploymentApi {
 
   @Autowired private OperationContext operationContext;
 
+  @Autowired private DeploymentServiceV2 deploymentServiceV2;
+
+  @Autowired private DataJobDeploymentPropertiesConfig dataJobDeploymentPropertiesConfig;
+
   @Override
   public ResponseEntity<Void> deploymentDelete(
       String teamName, String jobName, String deploymentId) {
     if (jobsService.jobWithTeamExists(jobName, teamName)) {
       // TODO: deploymentId not implemented
       if (jobName != null) {
-        deploymentService.deleteDeployment(jobName);
-        return ResponseEntity.accepted().build();
+        if (dataJobDeploymentPropertiesConfig
```

Comment:
I feel it's better this `if dataJobDeploymentPropertiesConfig.writeTo` it's in one place and not so many. 

And the appropriate DeploymentService is injected depending on dataJobDeploymentPropertiesConfig.writeTo value. 

So that may necessitate having DeploymentServiceInterface or something but that would be more maintainable, readable and easier to remove in the future if/when we decide to clean up.

Reply:
I am not sure how much easier it would be if we introduced an interface, the factory, and then, when we decide to clean up the code, reverted back to the same injection strategy. If you have a strong opinion I will refactor the code.

Reply:
ok